pkg/scheduler/objects/application.go

```diff
@@ -1742,3 +1754,10 @@ func (sa *Application) SetTimedOutPlaceholder(taskGroupName string, timedOut int
 		sa.placeholderData[taskGroupName].TimedOut = timedOut
 	}
 }
+
+// test only
+func (sa *Application) SetPreemptionAttemptInterval(interval time.Duration) {
```

Comment:
Again, I think we need to reconsider this. It's the RM's responsibility to kill tasks when we ask it to, and we need to trust that it will occur. However, expecting that this happens in some predetermined time is asking for trouble. By preempting once (only) and reserving the node, we are reserving space until the requisite tasks are terminated.

The only thing we might want to do is keep track of whether or not all the tasks have been preempted yet for a node. If they have, and we are still unschedulable, that likely indicates that another scheduler got in line ahead of us and we need to try preempting again.

Reply:
The interval is only considered if there was a preemption attempt for an ask, but no victim allocations were found. In this case, we probably want to do it again, but not immediately, because we'd flood the logs. If the preemption request was successful, then this interval doesn't matter.

Reply:
Maybe the naming `PreemptionRetryInterval` would be better?

Reply:
Yes, I like the rename idea, I think that's clearer.

Reply:
See earlier comment I think this is not needed at all.